Add database

i2GR opened the pull request.

i2GR committed:
новая ветка add-database
добавление ReadMe.md из sp 11.1

i2GR committed:
добавлен schema.sql

i2GR committed:
added DBUserService.java with 1 create query

i2GR committed:
interim (users ok)

i2GR committed:
users+friends+films
no genres

i2GR committed:
YP FP-11(2) add-database
initial
github-autocheck-fix1

i2GR committed:
YP FP-11(2) add-database
code-review fix1

i2GR committed:
YP FP-11(2) add-database
code-review fix1
(readme.md edit)
(clear png)

afomkina commented on src/main/java/ru/yandex/practicum/filmorate/controller/FriendsController.java:
```diff
@@ -31,36 +23,32 @@
 public class FriendsController {
 
     @NonNull
+    @Qualifier("userDBStorage")
```
Неверный qualifier

i2GR replied:
На @Qualifier IDEA ругается:

```Lombok does not copy the annotation 'org.springframework.beans.factory.annotation.Qualifier' into the constructor.```

Поэтому внедрение бинов было реализовано через @Primary, без которого ругался Spring-boot при запуске

```Consider marking one of the beans as @Primary, updating the consumer to accept multiple beans, or using @Qualifier to identify the bean that should be consumed```

Чтобы не писать конструкторы, я на текущий момент решил убрать не нужные @Qualifier-ы

afomkina commented on src/main/java/ru/yandex/practicum/filmorate/service/like/DBLikeService.java:
```diff
@@ -0,0 +1,57 @@
+     * хранилище в БД
+     */
+    @NonNull
+    @Autowired
```
`@Autowired` - лишний, так как инжект происходит с помощью `@RequiredArgsConstructor`

i2GR replied:
спасибо убрано. исправлено. и вроде как в голове окончательно упорядочено

afomkina reviewed (changes requested):
Отлично проработал запросы, молодец 🔥
Пометила пару опечаток, давай поправим 😉

i2GR committed:
YP FP-11(2) add-database
code-review fix2
(clear mistypes)

i2GR committed:
YP FP-11(2) add-database
code-review fix2
(clear mistypes)
(clear @Qualifiers (not needed. @Primary used@)
(Optional removed from FilmController#getTopFilms)

i2GR commented:
Спасибо за ревью 
...и с наступающим...
